packages/cli/bin/barnard59.js

```diff
@@ -31,8 +31,13 @@ const onError = async err => {
   process.off('SIGTERM', onError)
 
   if (err) {
-    // eslint-disable-next-line no-console
-    console.log(err)
+    if (err.skipTrace) {
+      // eslint-disable-next-line no-console
+      console.log(err.message)
+    } else {
```

Comment:
Yes, I would follow your suggestion and add a derived error class. Let's keep it in this module since it is responsible for this logic.

With that, I think you should move the construction of that message here 

```suggestion
    if (err instanceof MultipleRootsError) {
      const alternatives = err.pipelines.map(x => `\n\t--pipeline ${x}`).join('')
      // eslint-disable-next-line no-console
      console.log(`Multiple root pipelines found. Try one of these:${alternatives}`)
    } else {
```

And in `findPipeline.js` you'd simple throw like `throw new MultipleRootsError({ pipelines: ptr })`